homeassistant/components/html5/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for the html5 component."""
+
+import binascii
+from typing import cast
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import ec
+from py_vapid import Vapid
+from py_vapid.utils import b64urlencode
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import ATTR_VAPID_EMAIL, ATTR_VAPID_PRV_KEY, ATTR_VAPID_PUB_KEY, DOMAIN
+from .issues import async_create_html5_issue
+
+EMAIL_REGEX = r"^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$"
+
+
+def vapid_generate_private_key():
+    """Generate a VAPID private key."""
+    private_key = ec.generate_private_key(ec.SECP256R1(), default_backend())
+    return b64urlencode(
+        binascii.unhexlify(f"{private_key.private_numbers().private_value:x}".zfill(64))
+    )
+
+
+def vapid_get_public_key(private_key: str):
+    """Get the VAPID public key from a private key."""
+    vapid = Vapid.from_string(private_key)
+    public_key = cast(ec.EllipticCurvePublicKey, vapid.public_key)
+    return b64urlencode(
+        public_key.public_bytes(
+            serialization.Encoding.X962, serialization.PublicFormat.UncompressedPoint
+        )
+    )
+
+
+class HTML5ConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for HTML5."""
+
+    def _create_entry(
```

Comment:
Please mark this as a callback and prefix it with `async_`

https://developers.home-assistant.io/docs/asyncio_categorizing_functions#the-callback-function

Reply:
Done